web3sTests/Contract/ABIConstructorEncoderTests.swift

```diff
@@ -0,0 +1,25 @@
+//  Created by Rinat Enikeev on 24.04.2022.
+
+import XCTest
+import BigInt
+@testable import web3
+
+class ABIConstructorEncoderTests: XCTestCase {
+    var encoder: ABIConstructorEncoder!
+
+    override func setUp() {
+        encoder = ABIConstructorEncoder(TestConfig.erc20Bytecode.web3.hexData!)
+    }
+
+    // reference: https://ropsten.etherscan.io/address/0x583cbbb8a8443b38abcc0c956bece47340ea1367#code
+    func testGivenBokkyParams_ThenEncodesCorrectly() {
+        XCTAssertNoThrow(try encoder.encode("BokkyPooBah Test Token"))
+        XCTAssertNoThrow(try encoder.encode("BOKKY"))
+        XCTAssertNoThrow(try encoder.encode(UInt8(18)))
```

Comment:
Tests also serve as examples, can we create a test-only type to conform to `ABIConstructor` protocol? Also helps testing from API point of view not internal code.